dd-java-agent/instrumentation/play-2.6/src/main/java8/datadog/trace/instrumentation/play26/PlayHttpServerDecorator.java

```diff
@@ -66,12 +68,27 @@ public AgentSpan onRequest(final AgentSpan span, final Request request) {
     if (request != null) {
       // more about routes here:
       // https://github.com/playframework/playframework/blob/master/documentation/manual/releases/release26/migration26/Migration26.md
-      final Option<HandlerDef> defOption =
-          request.attrs().get(Router.Attrs.HANDLER_DEF.underlying());
-      if (!defOption.isEmpty()) {
-        final String path = defOption.get().path();
-        span.setTag(DDTags.RESOURCE_NAME, path);
+      Class HandlerDefClass = Router.Attrs.HANDLER_DEF.getClass();
+      Method asScalaOrUnderlying = null;
+
+      try {
+        asScalaOrUnderlying = HandlerDefClass.getMethod("underlying");
+      } catch (NoSuchMethodException e) {
+        try {
+          asScalaOrUnderlying = HandlerDefClass.getMethod("asScala");
+        } catch (NoSuchMethodException ee) {}
+      }
+
+      try {
+        final Option<HandlerDef> defOption =
+          request.attrs().get((TypedKey) asScalaOrUnderlying.invoke(Router.Attrs.HANDLER_DEF));
+        if (!defOption.isEmpty()) {
+          final String path = defOption.get().path();
+          span.setTag(DDTags.RESOURCE_NAME, path);
+        }
       }
+      catch (IllegalAccessException e) {}
+      catch (InvocationTargetException e) {}
```

Comment:
Same logging request

Reply:
Ideally you'd only need to wrap the invoke call and not the path access and tagging.

Reply:
`defOption` would be out of scope if I wrap only the invoke call.

Reply:
In that case you wouldn't mark it final and do a null check.